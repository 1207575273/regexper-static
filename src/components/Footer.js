import React from 'react';

const Footer = () => (
  <footer>
    <ul className="inline with-separator">
      <li>
        Created by <a href="mailto:jeff.avallone@gmail.com">Jeff Avallone</a>
      </li>
      <li>
        Generated images licensed: <a rel="license" href="http://creativecommons.org/licenses/by/3.0/">
          <img
            alt="Creative Commons CC-BY-3.0 License"
            src="https://licensebuttons.net/l/by/3.0/80x15.png" />
        </a>
      </li>
    </ul>
  </footer>
);

export default Footer;
