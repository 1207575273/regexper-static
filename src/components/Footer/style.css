@import url('../../globals.css');

footer {
  display: flex;
  align-items: flex-start;
  margin: var(--spacing-margin) 0;

  @media screen and (max-width: 800px) {
    display: block;
  }

  & ul {
    @apply --inline-list;
    @apply --with-separator-left;
    flex: 1;
  }

  & img {
    vertical-align: text-top;
    width: 80px;
    height: 15px;
  }

  & .buildId {
    color: color(var(--color-brown) blend(var(--color-tan) 25%));
  }
}
