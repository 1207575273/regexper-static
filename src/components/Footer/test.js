import React from 'react';
import { shallow } from 'enzyme';

import { Footer } from 'components/Footer';

describe('Footer', () => {
  test('rendering', () => {
    const component = shallow(
      <Footer>
        <p>Content</p>
      </Footer>
    );
    expect(component).toMatchSnapshot();
  });
});
