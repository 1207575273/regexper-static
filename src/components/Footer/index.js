import React from 'react';
import { translate, Trans } from 'react-i18next';

import style from './style.css';

const Footer = () => (
  <footer>
    <ul>
      <li>
        <Trans>Created by <a href="mailto:jeff.avallone@gmail.com">Jeff Avallone</a></Trans>
      </li>
      <li>
        <Trans i18nKey="Generated images licensed">
          Generated images licensed: <a rel="license external noopener noreferrer" target="_blank" href="http://creativecommons.org/licenses/by/3.0/">
            <img
              alt="Creative Commons CC-BY-3.0 License"
              src="https://licensebuttons.net/l/by/3.0/80x15.png" />
          </a>
        </Trans>
      </li>
      <li>
        <a href="/privacy.html"><Trans>Privacy Policy</Trans></a>
      </li>
    </ul>
    <div className={ style.buildId }>{ process.env.BUILD_ID }</div>
  </footer>
);

export default translate()(Footer);
export { Footer };
