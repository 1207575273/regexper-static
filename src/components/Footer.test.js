import React from 'react';
import { shallow } from 'enzyme';

import Footer from './Footer';

test('Footer rendering', () => {
  const component = shallow(
    <Footer>
      <p>Content</p>
    </Footer>
  );
  expect(component).toMatchSnapshot();
});
