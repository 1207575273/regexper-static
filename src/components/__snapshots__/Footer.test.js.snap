// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer rendering 1`] = `
<footer>
  <ul
    className="inline with-separator"
  >
    <li>
      <Trans>
        Created by 
        <a
          href="mailto:jeff.avallone@gmail.com"
        >
          Jeff Avallone
        </a>
      </Trans>
    </li>
    <li>
      <Trans
        i18nKey="Generated images licensed"
      >
        Generated images licensed: 
        <a
          href="http://creativecommons.org/licenses/by/3.0/"
          rel="license"
        >
          <img
            alt="Creative Commons CC-BY-3.0 License"
            src="https://licensebuttons.net/l/by/3.0/80x15.png"
          />
        </a>
      </Trans>
    </li>
  </ul>
</footer>
`;
